src/TesApi.Web/Storage/TerraStorageAccessProvider.cs

```diff
@@ -0,0 +1,161 @@
+                return await GetMappedSasContainerUrlFromWsmAsync(normalizedPath);
+            }
+
+            if (IsItKnownExecutionFilePath(normalizedPath))
+            {
+                return await GetMappedSasUrlFromWsmAsync(normalizedPath);
+            }
+
+            if (!StorageAccountUrlSegments.TryCreate(normalizedPath, out var segments))
+            {
+                throw new Exception(
+                    "Invalid path provided. The path must be a valid blob storage url or a path with the following format: /accountName/container");
+            }
+
+            CheckIfAccountAndContainerAreWorkspaceStorage(segments.AccountName, segments.ContainerName);
+
+            return await GetMappedSasUrlFromWsmAsync(segments.BlobName);
+        }
+
+        private async Task<string> GetMappedSasContainerUrlFromWsmAsync(string pathToAppend)
+        {
+            //an empty blob name gets a container Sas token
+            var tokenInfo = await GetSasTokenFromWsmAsync(CreateTokenParamsFromOptions(blobName: string.Empty, SasContainerPermissions));
+
+            var urlBuilder = new UriBuilder(tokenInfo.Url);
+
+            urlBuilder.Path += pathToAppend;
+
+            return urlBuilder.ToString();
+        }
+
+        private async Task<string> GetMappedSasUrlFromWsmAsync(string blobName)
+        {
+            var tokenParams = CreateTokenParamsFromOptions(blobName, SasBlobPermissions);
+
+            var tokenInfo = await GetSasTokenFromWsmAsync(tokenParams);
+
+            logger.LogInformation($"Successfully obtained the Sas Url from Terra. Requested blobName:{blobName}. Wsm resource id:{terraOptions.WorkspaceStorageContainerResourceId}");
+
+            return tokenInfo.Url;
+        }
+
+        private SasTokenApiParameters CreateTokenParamsFromOptions(string blobName, string sasPermissions)
+        {
+            return new SasTokenApiParameters(
+                terraOptions.SasAllowedIpRange,
+                terraOptions.SasTokenExpirationInSeconds,
+                sasPermissions, blobName);
+        }
+
+        private async Task<WsmSasTokenApiResponse> GetSasTokenFromWsmAsync(SasTokenApiParameters tokenParams)
+        {
+            logger.LogInformation(
+                $"Getting Sas Url from Terra. Requested blobName:{tokenParams.SasBlobName}. Wsm resource id:{terraOptions.WorkspaceStorageContainerResourceId}");
+
+            return await terraWsmApiClient.GetSasTokenAsync(
+                Guid.Parse(terraOptions.WorkspaceId),
+                Guid.Parse(terraOptions.WorkspaceStorageContainerResourceId),
+                tokenParams);
+        }
+
+        private void CheckIfAccountAndContainerAreWorkspaceStorage(string accountName, string containerName)
+        {
+            if (!IsTerraWorkspaceStorageAccount(accountName))
+            {
+                throw new Exception($"The account name does not match. Expected:{accountName} and Provided{terraOptions.WorkspaceStorageAccountName}");
```

Comment:
account name could contain customer information

Reply:
Okay. I see existing log entries that have this information. I will add an issue to look everywhere. I will remove this in my next PR for now.